numpy/core/code_generators/genapi.py

```diff
@@ -281,9 +281,8 @@ def find_functions(filename, tag='API'):
                     state = SCANNING
                 else:
                     function_args.append(line)
-        except Exception:
-            print(filename, lineno + 1)
-            raise
+        except Exception as e:
+            raise Exception(filename, lineno + 1) from e
```

Comment:
I think this should probably be:
```suggestion
        except ParseError:
            raise
        except Exception as e:
            raise ParseError(filename, lineno + 1, "see chained exception for details") from e
```

Reply:
Thank you for the insight-I just updated the code!